core/includes/install.core.inc

```diff
@@ -1454,6 +1450,341 @@ function install_select_language_form($form, &$form_state, $files) {
   return $form;
 }
 
+/**
+ * Provide a list of available translation files.
+ *
+ * @return array
+ *   An array of language codes.
+ */
+function install_get_available_translations($files) {
+  $available_translations = array();
+  $response = backdrop_http_request(INSTALL_AVAILABLE_TRANSLATIONS_URI);
+
+  $data = '';
+  if ($response->code === '200') {
+    $data = json_decode($response->data, TRUE);
+  }
+
+  if ($data) {
+    $available_translations = array_keys($data);
+  }
+  elseif (count($files) > 1) {
+    foreach($files as $translation_file) {
+      $available_translations[] = $translation_file->langcode;
+    }
+  }
+  else {
+    return FALSE;
```

Comment:
Better return the empty array instead?

Then we'd have a reliable return type. Empty array seems fine to me. Other thoughts?

Reply:
This last `else` should have been removed. We should always return at least English, as noted below.

And we shouldnt need a message, since we test the server is available in `install_check_localization_server()` before we call this, and that sets up the messages.

Reply:
Fixed